library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt

```diff
@@ -25,6 +25,22 @@ data class RetryPolicy(
       throw IllegalArgumentException("Per-retry timeout cannot be less than total timeout")
     }
   }
+
+  /**
+   * Initialize the retry policy from a set of headers.
+   *
+   * @param headers: The headers with which to initialize the retry policy.
+   */
+  constructor(headers: Headers): this(
+      headers.value("x-envoy-max-retries")?.first()!!.toInt(),
+      // TODO: should we have ? after the map?
```

Comment:
Contingent on https://github.com/lyft/envoy-mobile/pull/840/files#r424029491

Reply:
I guess my question is what optional is the `?` after map talking about? The first one is because the key `x-envoy-retry-on` might not be in the headers map so the list is optional. But the return type of map is `List<T>`, so I am not sure what ? means there.

Reply:
Yea I'm actually curious about this as well

Reply:
Are you talking about the elvis symbol `?:` here rather than the `?` symbol?